File: KCD/HMLocalDataStore.m
//
//  HMLocalDataStore.m
//  KCD
//
//  Created by Hori,Masaki on 2014/03/21.
//  Copyright (c) 2014年 Hori,Masaki. All rights reserved.
//

#import "HMLocalDataStore.h"

@implementation HMLocalDataStore

+ (void)load
{
	static dispatch_once_t onceToken;
	dispatch_once(&onceToken, ^{
		[self defaultManager];
	});
}
- (NSString *)modelName
{
	return @"LocalData";
}
- (NSString *)storeFileName
{
#if COREDATA_STORE_TYPE == 0
	return @"LocalData.storedata";
#else
	return @"LocalData.storedata.xml";
#endif
}
- (NSString *)storeType
{
#if COREDATA_STORE_TYPE == 0
	return NSSQLiteStoreType;
#else
	return NSXMLStoreType;
#endif
}
- (NSDictionary *)storeOptions
{
	NSDictionary *options = @{
							  NSMigratePersistentStoresAutomaticallyOption : @YES,
							  NSInferMappingModelAutomaticallyOption : @YES
							  };
	return options;
}
- (BOOL)deleteAndRetry
{
	return NO;
}

@end

File: KCD/HMLocalDataStore.swift
//
//  HMLocalDataStore.swift
//  KCD
//
//  Created by Hori,Masaki on 2015/01/02.
//  Copyright (c) 2015年 Hori,Masaki. All rights reserved.
//

import Foundation

private var options: [NSObject : AnyObject]? = nil
/**
サーバーサイドには存在しないデータを保存する
*/
class HMLocalDataStore: HMCoreDataManager {
	required init(type: HMCoreDataManagerType) {
		if options == nil {
			let param = NSDictionary(object: "MEMORY", forKey: "journal_mode")
			options = [
				NSSQLitePragmasOption : param,
				NSMigratePersistentStoresAutomaticallyOption : true,
				NSInferMappingModelAutomaticallyOption : true
			]
		}
		super.init(type: type)
	}
	
	override func modelName() -> String {
		return "LocalData"
	}
	override func storeFileName() -> String {
		return "LocalData.storedata"
	}
	override func storeType() -> String {
		return NSSQLiteStoreType
	}
	override func storeOptions() -> [NSObject : AnyObject]? {
		return options
	}
	override func deleteAndRetry() -> Bool {
		return false
	}
}
